segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/Write.java

```diff
@@ -60,12 +68,19 @@
         this.beginAttemptTimer = new AtomicReference<>();
     }
 
+    private ByteBuf convertData(CompositeArrayView data) {
```

Comment:
Is this the best place for this method?

Reply:
This method converts a `CompositeByteArraySegment` to a Netty ByteBuf. `CompositeByteArraySegment` is in `common` which doesn't have a dependency on Netty, so I can't do that there.

This is the main reason why I added the `collect` method in `CompositeArrayView`, so that I can essentially convert it to anything else without having to add extra dependencies in it.